Gems/Prefab/PrefabBuilder/PrefabGroup/DefaultProceduralPrefab.cpp

```diff
@@ -280,15 +281,14 @@ namespace AZ::SceneAPI
         }
 
         // assign mesh asset id hint using JSON
-        AZStd::string modelAssetPath;
-        modelAssetPath = relativeSourcePath;
-        AZ::StringFunc::Path::ReplaceFullName(modelAssetPath, meshGroupName.c_str());
-        AZ::StringFunc::Replace(modelAssetPath, "\\", "/"); // asset paths use forward slashes
+        AZ::IO::Path modelAssetPath(relativeSourcePath, '/');
+        modelAssetPath.ReplaceFilename(AZ::IO::PathView(meshGroupName));
+        modelAssetPath.ReplaceExtension(AZ::IO::PathView(sourceFileExtension));
 
         auto meshAssetJson = AZStd::string::format(
             R"JSON(
                    {"Controller": {"Configuration": {"ModelAsset": { "assetHint": "%s.azmodel"}}}}
-             )JSON", modelAssetPath.c_str());
+             )JSON", modelAssetPath.LexicallyNormal().String().c_str());
```

Comment:
Nitpick: You could use `FixedString()` instead of `String()` to prevent a heap allocation to create a temporary `AZStd::string`
```suggestion
             )JSON", modelAssetPath.LexicallyNormal().FixedString().c_str());
```

Reply:
I didn't think FixedMaxPathStringAsPosix did the same thing as LexicallyNormal though? LexicallyNormal will condense down any excess paths and separators. Admittedly maybe it's overkill here, it's possible that we can never get _bad_ paths here, but it seemed like a good idea to clean the path just in case.

Reply:
You are right that `FixedMaxPathStringAsPosix` doesn't collapse any "." or ".." separators down on the path.

I was assuming that wouldn't be an issue. 

Any paths that are compared using the `AZ::IO::Path*` API don't have to be string equivalent to [compare equal](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.inl#L382-L403), so normalization is only needed if the desire is to perform raw string comparison of the path data which I don't recommend as the Path API is better for comparison.
Comparing `ambulance/truck/../model.fbx` against `ambulance/model.fbx` would compare equal using the Path API

Or the second case is to normalize the serialized output when serializing a path which I do recommend. Though in that case I would recommend a normalizing the path and then converting to a Posix path, so that both Windows and non-Windows produce the same paths

Reply:
This particular chunk of code is directly writing out the asset paths into the .procprefab json file, which is why I was more concerned with cleaning the path here. It just seems like a good idea to write out the "best" version of the path that we have at serialization time.